ItemList.tsx

```diff
@@ -0,0 +1,114 @@
+import { useState } from 'react';
+import { Alert, FlatList, Text, TextInput, TouchableOpacity, View } from 'react-native';
+import { ItemCard } from './ItemCard';
+import { StyleSheet } from 'react-native';
+import { ImageData } from './ItemCard';
+import { Dimensions } from 'react-native';
+import ImageModal from './ImageModal';
+
+const numColumns = 3;
+const screenWidth = Dimensions.get('window').width;
+const itemWidth = screenWidth / numColumns;
+const itemHeight = itemWidth * 2;
+
+const PICS: ImageData[] = [];
+for (let i = 1; i < 70; i++) {
+  PICS.push({ id: i, url: `https://picsum.photos/id/${i}/200`, width: itemWidth, height: itemHeight});
+}
+
+
+
+export const ItemList = () => 
+{
+    const [searchQuery, setSearchQuery] = useState('');
+    const [activeItems, setActiveItems] = useState(PICS);
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const [selectedImageUrl, setSelectedImageUrl] = useState('');
+
+    const handleOnPress = (url: string) => {
+        setIsModalVisible(true);
+        setSelectedImageUrl(url);
+      };
+
+    const renderItem = ({ item }: any) => (
+        <TouchableOpacity onPress={() => handleOnPress(item.id)}>
+            <ItemCard id={item.id} url={item.url} width={itemWidth} height={itemHeight} />
+        </TouchableOpacity>
+    );
+
+    const search = (query: string) => {
+        const filteredItems = PICS.filter((contact) => 
+        contact.id.toString().toLowerCase().includes(query.toLowerCase()));
+        setActiveItems(filteredItems);
+        setSearchQuery(query);
+    }
+
+    return (
+        <View style={styles.container}>
+            <TextInput 
+                value={searchQuery}
+                onChangeText={search}
+                placeholder='Search Here'
+                style={styles.input}
+            />
+
+            <FlatList 
+                data={activeItems}
+                keyExtractor={(item) => item.id.toString()}
+                renderItem={({ item }) => (
+                    <TouchableOpacity onPress={() => handleOnPress(item.url)}>
```

Comment:
This JSX within `renderItem` is a duplicate of the `renderItem` function.

It could look like this
```
<FlatList
   renderItem={renderItem
...other props
/>
```